Extdn/Sniffs/Blocks/DeprecatedParentsSniff.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © ExtDN. All rights reserved.
+ */
+
+declare(strict_types=1);
+
+namespace Extdn\Sniffs\Blocks;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use Extdn\Utils\Reflection;
+use Zend\Server\Reflection\ReflectionClass;
+
+/**
+ * Class DeprecatedParentsSniff
+ *
+ * @package Extdn\Sniffs\Classes\Constructor
+ */
+class DeprecatedParentsSniff implements Sniff
+{
+    /**
+     * @var string
+     */
+    protected $message = 'The ObjectManager should not be injected into the constructor';
+
+    /**
+     * @inheritdoc
+     */
+    public function register()
+    {
+        return [T_CLASS];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $className = Reflection::findClassName($phpcsFile);
+        if (empty($className)) {
+            return false;
+        }
+
+        // Make sure to load the file itself, so that autoloading can be skipped
+        include_once($phpcsFile->getFilename());
+
+        $class = Reflection::getClass($className);
+        $parentClass = $class->getParentClass();
+
+        foreach ($this->getDeprecatedClasses() as $deprecatedClass) {
+            if ($parentClass->getName() !== $deprecatedClass['class']) {
+                continue;
+            }
+
+            $warning = sprintf('Block parent "%s" is deprecated. %s', $deprecatedClass['class'], $deprecatedClass['advice']);
+            $phpcsFile->addWarning($warning, null, 'deprecated-parent');
+        }
+    }
+
+    /**
+     * @return array
+     */
+    private function getDeprecatedClasses(): array
+    {
+        return [
+            [
+                'class' => 'Magento\Backend\Block\Widget\Form\Generic',
+                'advice' => 'Refactor this to a UiComponent.'
```

Comment:
I think it will not be clear for people who just started working with magento. I think it's better to add there any link to 
 explanation "why?" with example how to refactor it

Reply:
We use a standardized URL to document sniffs in the repo (it would be `Extdn/Sniffs/Blocks/DeprecatedParentsSniff.md`, but is missing from this PR)

But good question: shouldn't we show this URL in the messages?

Reply:
Doc has now been added: https://github.com/jissereitsma/extdn-phpcs/commit/f831df3134ee6a3f3b2707ebd11b31e218311668 And the rule warning now refers to the actual URL (which is not there yet): https://github.com/jissereitsma/extdn-phpcs/commit/078417c1bf148d9214cc34a337eaf41162096ba9